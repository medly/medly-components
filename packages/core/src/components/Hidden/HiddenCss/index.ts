export { HiddenCss as default } from './HiddenCss.styled';
