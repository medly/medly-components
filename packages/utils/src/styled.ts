import * as styledComponents from 'styled-components';

const { default: styled, css, createGlobalStyle, keyframes, ThemeProvider, withTheme } = styledComponents;

export { styled, css, createGlobalStyle, keyframes, ThemeProvider, withTheme };
