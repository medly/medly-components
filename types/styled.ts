import 'styled-components';
import { Theme } from '../packages/theme';

declare module 'styled-components' {
    export interface DefaultTheme extends Theme {}
}
