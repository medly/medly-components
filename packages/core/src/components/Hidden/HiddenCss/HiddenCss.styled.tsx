import { breakpoints, css, InjectClassName, media, styled } from '@medly-components/utils';
import { HiddenProps } from '../types';

const hide = css`
    display: none;
`;

export const HiddenCss = styled(InjectClassName)<HiddenProps>`
    ${({ theme, up }) =>
        up &&
        media(breakpoints(theme.breakpoints).up(up))`
            display: none;
        `}
    ${({ theme, down }) =>
        down &&
        media(breakpoints(theme.breakpoints).down(down))`
            display: none;
        `}
    ${({ theme, between }) =>
        between &&
        media(breakpoints(theme.breakpoints).between(between.start, between.end))`
            display: none;
        `}
    ${({ theme, only }) =>
        only &&
        media(breakpoints(theme.breakpoints).only(only))`
            display: none;
        `}
    ${({ theme, multiple }) =>
        multiple &&
        media(breakpoints(theme.breakpoints).multiple(multiple))`
            display: none;
        `}
    
        ${({ up, down, between, only, multiple }) => !up && !down && !between && !only && !multiple && hide}
`;
HiddenCss.displayName = 'HiddenCss';
