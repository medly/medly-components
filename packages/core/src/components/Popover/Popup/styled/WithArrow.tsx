import { css } from 'styled-components';
import { PopoverPopupProps } from '../types';

const getPositionWithArrow = ({ placement }: PopoverPopupProps) => {
    switch (placement) {
        case 'top':
            return css`
                box-shadow: 0 0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateX(-50%);
                    top: 100%;
                    left: 50%;
                }
            `;
        case 'top-start':
            return css`
                box-shadow: 0 0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    top: 100%;
                    left: 2rem;
                }
            `;
        case 'top-end':
            return css`
                box-shadow: 0 0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    top: 100%;
                    right: 2rem;
                }
            `;
        case 'right':
            return css`
                box-shadow: -0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateY(-50%) rotate(90deg);
                    top: 50%;
                    right: calc(100% - 0.3rem);
                }
            `;
        case 'right-start':
            return css`
                box-shadow: -0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(90deg);
                    top: 2rem;
                    right: calc(100% - 0.3rem);
                }
            `;
        case 'right-end':
            return css`
                box-shadow: -0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(90deg);
                    bottom: 2rem;
                    right: calc(100% - 0.3rem);
                }
            `;
        case 'bottom':
            return css`
                box-shadow: 0 -0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateX(-50%) rotate(180deg);
                    left: 50%;
                    bottom: 100%;
                }
            `;
        case 'bottom-start':
            return css`
                box-shadow: 0 -0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(180deg);
                    left: 2rem;
                    bottom: 100%;
                }
            `;
        case 'bottom-end':
            return css`
                box-shadow: 0 -0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(180deg);
                    right: 2rem;
                    bottom: 100%;
                }
            `;
        case 'left':
            return css`
                box-shadow: 0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateY(-50%) rotate(270deg);
                    top: 50%;
                    left: calc(100% - 0.3rem);
                }
            `;
        case 'left-start':
            return css`
                box-shadow: 0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(270deg);
                    top: 2rem;
                    left: calc(100% - 0.3rem);
                }
            `;
        case 'left-end':
            return css`
                box-shadow: 0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(270deg);
                    bottom: 2rem;
                    left: calc(100% - 0.3rem);
                }
            `;
    }
};

export const getStyleWithArrow = css<PopoverPopupProps>`
    border-radius: 0.8rem;
    padding: 1.5rem 2rem;

    &::after {
        content: '';
        width: 0;
        height: 0;
        z-index: 4;
        position: absolute;
        border-left: 0.8rem solid transparent;
        border-right: 0.8rem solid transparent;
        border-top: 1rem solid ${({ bgColor }) => bgColor};
    }

    ${getPositionWithArrow}
`;
