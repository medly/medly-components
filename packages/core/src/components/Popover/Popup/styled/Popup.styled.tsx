import { fullHeight, fullWidth } from '@medly-components/utils';
import styled, { css } from 'styled-components';
import { Placement } from '../../types';
import { StyledPopoverPopupProps } from '../types';
import { getStyleWithArrow } from './WithArrow';

export const getPosition = ({ placement, distance = '0px' }: { placement: Placement; distance?: string }) => {
    switch (placement) {
        case 'top-start':
            return css`
                bottom: calc(100% + ${distance});
                left: 0%;
            `;
        case 'top':
            return css`
                bottom: calc(100% + ${distance});
                left: 50%;
                transform: translateX(-50%);
            `;
        case 'top-end':
            return css`
                bottom: calc(100% + ${distance});
                right: 0%;
            `;
        case 'right-start':
            return css`
                top: 0%;
                left: calc(100% + ${distance});
            `;
        case 'right':
            return css`
                left: calc(100% + ${distance});
                top: 50%;
                transform: translateY(-50%);
            `;
        case 'right-end':
            return css`
                bottom: 0;
                left: calc(100% + ${distance});
            `;
        case 'bottom-start':
            return css`
                top: calc(100% + ${distance});
                left: 0%;
            `;
        case 'bottom':
            return css`
                top: calc(100% + ${distance});
                left: 50%;
                transform: translateX(-50%);
            `;
        case 'bottom-end':
            return css`
                top: calc(100% + ${distance});
                right: 0%;
            `;
        case 'left-start':
            return css`
                right: calc(100% + ${distance});
                top: 0%;
            `;
        case 'left':
            return css`
                right: calc(100% + ${distance});
                top: 50%;
                transform: translateY(-50%);
            `;
        case 'left-end':
            return css`
                right: calc(100% + ${distance});
                bottom: 0%;
            `;
    }
};

export const PopupStyled = styled('div')<StyledPopoverPopupProps>`
    z-index: 4;
    position: absolute;
    background-color: ${({ bgColor }) => bgColor};

    ${props => props.fullWidth && fullWidth()};
    ${props => props.fullHeight && fullHeight()};

    ${props => getPosition({ placement: props.placement, distance: props.withArrow ? '1.5rem' : props.distance })}
    ${({ withArrow }) => withArrow && getStyleWithArrow}
`;
