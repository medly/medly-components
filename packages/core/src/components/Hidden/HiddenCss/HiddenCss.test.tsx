import { render } from '@test-utils';
import { HiddenCss } from './HiddenCss.styled';

describe('HiddenCss', () => {
    test.each([
        ['at or above', { up: 'S' }],
        ['at or down', { down: 'S' }],
        ['between start and end of', { between: { start: 'S', end: 'L' } }],
        ['at specific', { only: 'S' }],
        ['at the specified list of', { multiple: ['M', 'XL'] }]
    ])('should hide the children %s breakpoint', (propKey: any, props: any) => {
        const { container } = render(
            <HiddenCss {...props}>
                <span>Strong text</span>
            </HiddenCss>
        );
        expect(container).toMatchSnapshot();
    });

    it('should hide the children if none of the prop is passed', () => {
        const { container } = render(
            <HiddenCss>
                <span>Strong text</span>
            </HiddenCss>
        );
        expect(container).toMatchSnapshot();
    });
});
