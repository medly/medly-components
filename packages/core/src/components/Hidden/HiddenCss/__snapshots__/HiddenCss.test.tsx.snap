// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HiddenCss should hide the children at or above breakpoint 1`] = `
@media (min-width: 0px) {
  .c0 {
    display: none;
  }
}

<div>
  <span
    class="c0"
  >
    Strong text
  </span>
</div>
`;

exports[`HiddenCss should hide the children at or down breakpoint 1`] = `
@media (max-width: 767px) {
  .c0 {
    display: none;
  }
}

<div>
  <span
    class="c0"
  >
    Strong text
  </span>
</div>
`;

exports[`HiddenCss should hide the children at specific breakpoint 1`] = `
@media (min-width: 0px) and (max-width: 767px) {
  .c0 {
    display: none;
  }
}

<div>
  <span
    class="c0"
  >
    Strong text
  </span>
</div>
`;

exports[`HiddenCss should hide the children at the specified list of breakpoint 1`] = `
@media (min-width: 768px) and (max-width: 1023px),(min-width: 1280px) and (max-width: 2400px) {
  .c0 {
    display: none;
  }
}

<div>
  <span
    class="c0"
  >
    Strong text
  </span>
</div>
`;

exports[`HiddenCss should hide the children between start and end of breakpoint 1`] = `
@media (min-width: 0px) and (max-width: 1279px) {
  .c0 {
    display: none;
  }
}

<div>
  <span
    class="c0"
  >
    Strong text
  </span>
</div>
`;

exports[`HiddenCss should hide the children if none of the prop is passed 1`] = `
.c0 {
  display: none;
}

<div>
  <span
    class="c0"
  >
    Strong text
  </span>
</div>
`;
